addons/redis/dataprotection/pitr-backup.sh

```diff
@@ -0,0 +1,151 @@
+  tar -cvf - "$(generate_backup_manifest)" -C "${DATA_DIR}" \
+    "${AOF_DIR}/$(basename "${incr_file}")" "${AOF_DIR}/$(basename "${base_file}")" \
+    "users.acl" | datasafed push -z zstd - "${target_file}"
+
+  # delete remote and local uncompressed files we are tracking.
+  datasafed rm -r "${backup_files_prefix}.dir"
+  rm "${incr_file}" "${base_file}"
+
+  DP_log "Archived files: ${incr_file} and ${base_file} to $target_file"
+}
+
+function update_aof_file() {
+  local incr_file=$(aof_incr_file "$global_backup_seq")
+  local base_file=$(aof_base_file "$global_backup_seq")
+  local backup_files_prefix=$(get_backup_files_prefix $base_file)
+  local backup_manifest="$(generate_backup_manifest)"
+
+  # create a directory for backup files we are tracking, and after a aof rewrite, we will archive them in to a tar.zst file
+  if [ $(stat -c %Y "${base_file}") -gt ${global_aof_last_modify_time} ]; then
+    datasafed push "${base_file}" "${backup_files_prefix}.dir/${AOF_DIR}/$(basename "${base_file}")"
+    datasafed push "${backup_manifest}" "${backup_files_prefix}.dir/${backup_manifest}"
+    datasafed push "${DATA_DIR}/users.acl" "${backup_files_prefix}.dir/users.acl"
+    DP_log "Upload file: $base_file users.acl"
+  fi
+
+  # keep updating the latest aof file
+  local aof_modify_time=$(stat -c %Y "${incr_file}")
+  if [ "${aof_modify_time}" -gt "${global_aof_last_modify_time}" ]; then
+    datasafed push "${incr_file}" "${backup_files_prefix}.dir/${AOF_DIR}/$(basename "${incr_file}")"
+    global_aof_last_modify_time=${aof_modify_time}
+    DP_log "Update file: $incr_file"
+  fi
+}
+
+function purge_expired_files() {
+  local current_unix=$(date +%s)
+  info=$(DP_purge_expired_files ${current_unix} ${global_last_purge_time} / 600)
+  if [ ! -z "${info}" ]; then
+    global_last_purge_time=${currentUnix}
+    DP_log "Cleanup expired aof files: ${info}"
+    local total_size=$(datasafed stat / | grep TotalSize | awk '{print $2}')
+    DP_save_backup_status_info "${total_size}"
+  fi
+}
+
+function set_conf() {
+  ${connect_url} CONFIG SET appendonly yes
+  ${connect_url} CONFIG SET aof-disable-auto-gc yes
+  ${connect_url} CONFIG SET aof-timestamp-enabled yes
+}
+
+function save_backup_status() {
+  # if no size changes, return
+  local total_size=$(datasafed stat / | grep TotalSize | awk '{print $2}')
+  if [[ -z ${total_size} || ${total_size} -eq 0 || ${total_size} == ${global_old_size} ]]; then
+    return
+  fi
+  global_old_size=${total_size}
+  local start_time=$(datasafed list / | awk -F '.' '{print $1}' | sort | head -n 1)
+  DP_save_backup_status_info "${total_size}" "${start_time}" "$(date +%s)"
+}
+
+trap "echo 'Terminating...' && sync && ${connect_url} CONFIG SET aof-disable-auto-gc no && exit 0" TERM
+echo "INFO: start to backup"
+set_conf
+while true; do
+  aof_seq=$(awk '/type i/ { print $4 }' "${AOF_MANIFEST_FILE}")
+  while [ "${global_backup_seq}" -lt "${aof_seq}" ]; do
+    archive_pair_files
```

Comment:
There are multiple instances in the  replication mode, and the AOF files of each instance are inconsistent. It is necessary to archive and backup each pod